### PR TITLE
[VIVO-1436] Implementation of Advanced Role Management

### PR DESCRIPTION
**Pull request by grahamtriggs**
Reopening as old pull request was incorrect

See https://github.com/vivo-project/Vitro/pull/80 for details

### CODE REVIEW

File: api/src/main/java/edu/cornell/mannlib/vivo/auth/policy/SelfEditorRelationship.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	/**
+	 * When the system starts up, install the policy. This class must be a
+	 * listener in web.xml
+	 * 
+	 * The CommonIdentifierBundleFactory already creates the IDs we need.
+	 */
+	public static class Setup implements ServletContextListener {
+		@Override
+		public void contextInitialized(ServletContextEvent sce) {
+			ServletContext ctx = sce.getServletContext();
```
Comment (awoods) [inline api/src/main/java/edu/cornell/mannlib/vivo/auth/policy/SelfEditorRelationship.java]: The `ctx` variable does not appear to be used... and can likely be removed.

File: webapp/src/main/webapp/WEB-INF/resources/startup_listeners.txt [hunk 2]
```diff
@@ -33,6 +33,9 @@ edu.cornell.mannlib.vitro.webapp.servlet.setup.FileGraphSetup
 # edu.cornell.mannlib.vitro.webapp.migration.rel17.Release17Migrator
 edu.cornell.mannlib.vitro.webapp.migration.rel18.Release18Migrator
 
+edu.cornell.mannlib.vitro.webapp.migration.auth.AuthMigrator
```
Comment (awoods) [inline webapp/src/main/webapp/WEB-INF/resources/startup_listeners.txt]: Please add a comment describing this migrator.

**Comment by grahamtriggs** [comment]
@j2blake The issues with leftover roles in the ontology files was a deliberate choice - it is relatively easy (albeit slightly time consuming) to delete lines that are no longer required. However, not knowing how long it will take to review the role management, or what might happen in ontology development in that time, I was hoping to reduce potential merge problems by delaying removing the old permissions.

**Comment by grahamtriggs** [comment]
@j2blake About the nested class in SelfEditorRelationship - it was done for consistency with other setup code that is called from startup_listeners.txt. I've added a comment to the class.

File: webapp/src/main/webapp/WEB-INF/resources/startup_listeners.txt (hunk 2 above)
Comment (awoods) [reply webapp/src/main/webapp/WEB-INF/resources/startup_listeners.txt]: resolved

**Comment by gneissone** [comment]
Reopening this pull request. In an effort to follow current best practices, the VIVO project will now be developed against the `master` branch. As a result of removing the `develop` branch, this pull request was automatically closed by GitHub, apologies for any confusion this may have caused.

**Comment by litvinovg** [comment]
Superseded by https://github.com/vivo-project/VIVO/pull/3887

**Pull request closed without merging**